eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/jpa/AggregateBasedEventEntry.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore.jpa;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import jakarta.persistence.Basic;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Index;
+import jakarta.persistence.Lob;
+import jakarta.persistence.SequenceGenerator;
+import jakarta.persistence.Table;
+import org.axonframework.common.DateTimeUtils;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.MessageType;
+
+import java.time.Instant;
+import java.time.temporal.TemporalAccessor;
+
+import static org.axonframework.common.DateTimeUtils.formatInstant;
+
+/**
+ * A JPA entry dedicated for storing {@link EventMessage EventMessages} in the
+ * {@link AggregateBasedJpaEventStorageEngine}.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+@Entity
+@Table(indexes = @Index(columnList = "aggregateIdentifier,aggregateSequenceNumber", unique = true))
+public class AggregateBasedEventEntry {
+
+    // Deliberate field ordering for column ordering
+    @Id
+    @GeneratedValue(
+            strategy = GenerationType.SEQUENCE,
+            generator = "globalIndexGenerator"
+    )
+    @SequenceGenerator(
+            name = "globalIndexGenerator",
+            sequenceName = "aggregate-based-event-global-index-sequence",
+            allocationSize = 1
+    )
+    private long globalIndex;
+    private String aggregateType;
+    private String aggregateIdentifier;
+    private Long aggregateSequenceNumber;
+    @Basic(optional = false)
+    private String type;
+    @Basic(optional = false)
+    private String version;
+    @Basic(optional = false)
+    private String timestamp;
+    @Basic(optional = false)
+    @Lob
+    private byte[] payload;
+    @Lob
+    private byte[] metaData;
```

Comment:
You need to adjust the name of the column to `metadata` or change it to `metaData` in SQL queries in `LegacyJpaEventStorageOperations`. 

I assume you were changing it back and forth with refactor and IntelliJ has found it in strings or something :D

Reply:
Sharp eye as always :see_no_evil: 
The problem should be fixed now. Thanks for nudging me, @MateuszNaKodach!